Source/SwiftyDropbox/Shared/Handwritten/OAuth/OAuth.swift

```diff
@@ -201,42 +197,71 @@ open class DropboxOAuthManager {
         return false
     }
 
+    /// Handles redirect URL from web.
+    /// Auth results are passed back in URL query parameters.
+    /// - Error result parameters looks like this:
+    /// ```
+    /// [
+    ///     "error": "<error_name>",
+    ///     "error_description: "<error_description>"
+    /// ]
+    /// ```
+    /// - Success result looks like these:
+    ///     1. Code flow result
+    ///     ```
+    ///     [
+    ///         "state": "<state_string>",
+    ///         "code": "<oauth_code>"
+    ///     ]
+    ///     ```
+    ///     2. Token flow result
+    ///     ```
+    ///     [
+    ///         "state": "<state_string>",
+    ///         "access_token": "<oauth2_access_token>",
+    ///         "uid": "<uid>"
+    ///     ]
+    ///     ```
     func extractFromRedirectURL(_ url: URL, completion: @escaping DropboxOAuthCompletion) {
-        let results = OAuthUtils.extractParamsFromUrl(url)
-        if let error = results["error"] {
+        let parametersMap = OAuthUtils.extractParamsFromUrl(url)
+        // Error case
+        if let error = parametersMap[OAuthConstants.errorKey] {
             let result: DropboxOAuthResult
-            let desc = results[OAuthConstants.errorDescription]?
-                .replacingOccurrences(of: "+", with: " ")
-                .removingPercentEncoding
-            if results[OAuthConstants.errorKey] != "access_denied" {
+            if error == OAuth2Error.accessDenied.rawValue {
+                // Treat as cancelled if user denied auth.
                 result = .cancel
             } else {
-                result = .error(OAuth2Error(errorCode: error), desc ?? "")
+                let description = parametersMap[OAuthConstants.errorDescription]
+                result = .error(OAuth2Error(errorCode: error), description)
             }
             completion(result)
-        } else {
-            let state = results[OAuthConstants.stateKey]
-            let storedState = UserDefaults.standard.string(forKey: Constants.kCSRFKey)
+            return
```

Comment:
I think it's clearer to have an `else` clause rather than this `return` for the failure vs success case.